lib/rules/no-throw-literal.js

```diff
@@ -1,12 +1,32 @@
 /**
  * @fileoverview Rule to restrict what can be thrown as an exception.
  * @author Dieter Oberkofler
+ * @copyright 2015 Ian VanSchooten. All rights reserved.
  * @copyright 2015 Dieter Oberkofler. All rights reserved.
  */
 
 "use strict";
 
 //------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Determine if a node is considered literal (including concatenation)
+ * @param  {ASTNode}  node  ASTNode to check for only literals
+ * @returns {Boolean}       True if considered literal, false otherwise
+ */
+function isLiteral(node) {
+    if (node.type === "Literal") {
+        return true;
+    }
+
+    if (node.type === "BinaryExpression") {
+        return true;
```

Comment:
The reason this came up was that someone had the following type of code which was not being caught:

```
if (counter !== 1) { throw 'some assertion failed to exec. counter = ' + counter } 
```

Am I wrong in thinking that **all** BinayExpressions will become a literal when executed?  If that is true, we can say the rule is to prevent throwing a literal *or things which are always guaranteed to be a literal*.  I think that still fits the name and is within the realm of static analysis (as long as they are true guarantees, and not depending on runtime interpretation).